src/core/chunked_stream.js

```diff
@@ -285,6 +285,12 @@ class ChunkedStream {
       }
       return missingChunks;
     };
+    ChunkedStreamSubstream.prototype.allChunksLoaded = function() {
+      if (this.numChunksLoaded === this.numChunks) {
+        return true;
+      }
+      return this.getMissingChunks().length === 0;
```

Comment:
Why is this check still needed here, i.e., why isn't the check above sufficient?

Reply:
The first check is essentially just a fast-path, but not *necessarily* a required condition for considering a `ChunkedStreamSubstream` instance as completely loaded.

Please keep in mind that we're inheriting from a `ChunkedStream` instance above, note in particular the special `getMissingChunks` method used here, hence why only a partially loaded `ChunkedStream` can still be a fully loaded `ChunkedStreamSubstream`.

Hopefully this all makes sense, since it does look a bit weird at first!

Reply:
Ah, the special `getMissingChunks` method is what I overlooked here. Thanks!